src/buildApplication/generateFunctionsMap.ts

```diff
@@ -436,3 +433,25 @@ type LooseNode = Node & {
   */
 	value: any;
 };
+
+function getChunkIdentifier(chunkKey: number): string {
+	return `__chunk_${chunkKey}`;
+}
+
+function getChunkImportFn(functionPath: string): (chunkKey: number) => string {
+	const functionNestingLevel = getFunctionNestingLevel(functionPath);
+	const accountForNestingPath = `../`.repeat(functionNestingLevel);
+	return chunkKey => {
+		const chunkIdentifier = getChunkIdentifier(chunkKey);
+		const chunkPath = `${accountForNestingPath}__next-on-pages-dist__/chunks/${chunkKey}.js`;
+		return `import ${chunkIdentifier} from '${chunkPath}'`;
+	};
+}
+
+function getFunctionNestingLevel(functionPath: string): number {
+	const relativePath = functionPath.replace(
+		/^.*\/.vercel\/output\/functions\//,
+		''
+	);
+	return relativePath.split('/').length;
+}
```

Comment:
That's some pretty intense path resolution stuff going on there. I think you could avoid this if you were to pre-compile the template, but no biggie.